02 markdown

JulianKarlBauer opened the pull request.

JulianKarlBauer committed:
Add gitlab-Specific-cheat-sheet. Differences e.g. Latex, ...

JulianKarlBauer committed:
Add challenge: Headings, Emphasis, Lists, Enumerations

Is this challenge too long?
Are all features relevant?
Specifically:
- Nesting
- Counting of enumerations

JulianKarlBauer committed:
Add callout on flavors: Which flavor does jekyll (Kramdown) use?

JulianKarlBauer committed:
Minor changes... is it possible to reduce redundancy

JulianKarlBauer committed:
Separate lists and enumeration from first challenge.

Note: The challenge on lists and enumerations may be deleted completely, 
because both are not part of the teaching skills.

JulianKarlBauer committed:
Add challenge on code: inline and block without syntax

JulianKarlBauer committed:
Add challenge on syntax highlighting. Issue: Rendering Python

Issue: How can the Python code be rendered without using the 
lesson-style?

tobyhodges commented:
This is great, @JulianKarlBauer! Reviewing now...

tobyhodges commented on _episodes/02-markdown.md:
```diff
@@ -24,9 +24,220 @@ keypoints:
+>
+> ~~~
+> Discussing code details can be very helpful.  
+> Therefore, referring to variables, e.g. _path_ in a formatted way is great.
```
```suggestion
> When doing so, it's great to be able to refer to variables, functions, etc, e.g. `path` in a formatted way.
```

tobyhodges commented on _episodes/02-markdown.md:
```diff
@@ -24,9 +24,220 @@ keypoints:
+> > ## Solution
+> > ~~~
+> > Discussing code details can be very helpful.  
+> > Therefore, referring to variables, e.g. `path` in a formatted way is great.
```
```suggestion
> > When doing so, it's great to be able to refer to variables, functions, etc, e.g. `path` in a formatted way.
```

tobyhodges commented on _episodes/02-markdown.md:
```diff
@@ -24,9 +24,220 @@ keypoints:
+
+> ## Syntax Highlighting
+> Syntax highlighting increases readability a lot.  
+> Guess in which programming language the following code snippets are written
```
```suggestion
> Below are code snippets written in HTML, Python, and R.
> Guess in which programming language each snippet is written
```

I think the above makes the exercise easier for those coming to the lesson without direct experience of Python/R

tobyhodges commented on _episodes/02-markdown.md:
```diff
@@ -24,9 +24,220 @@ keypoints:
+> > these line breaks  
+> > are caused by two trailing spaces at the end of the lines.
+> > ~~~
+> > {: .markdown }
```
the lesson template (currently) has no highlighting rules for Markdown source. Best to change all these `{: .markdown }` tags to `{: .source }`

JulianKarlBauer replied:
See 5c040eb

tobyhodges commented on _episodes/02-markdown.md:
````diff
@@ -24,9 +24,220 @@ keypoints:
+> and fill the gaps to activate syntax highlighting creating this rendered view
+> 
+> -------------
+> ```R
````
```suggestion
> ```r
```
code block highlighting tags should be lowercase

tobyhodges commented on _episodes/02-markdown.md:
```diff
@@ -24,9 +24,220 @@ keypoints:
+
+
+
+
```
It would be great to add one more exercise about links here, perhaps a fill-in-the-blanks? No need to do that now, though, we can open an Issue/include a TODO to remind us later.

JulianKarlBauer replied:
Opened an [issue](https://github.com/tobyhodges/building-websites-with-jekyll-and-github-or-gitlab/issues/4#issue-668666187) and will resolve it.

tobyhodges commented on _episodes/02-markdown.md:
```diff
@@ -24,9 +24,220 @@ keypoints:
+
+> ## Flavors
+>
+> Basic Markdown features are the same on all platforms such as Github and Gitlab.
```
```suggestion
> Basic Markdown features are the same on all platforms such as GitHub and GitLab.
```

tobyhodges commented on _episodes/02-markdown.md:
```diff
@@ -24,9 +24,220 @@ keypoints:
+> Basic Markdown features are the same on all platforms such as Github and Gitlab.
+> However, there are differences in the details and scope of the features offered.
+> To address this situation, the Markdown 
+> used at Github is called **Github-flavored Markdown**
```
```suggestion
> used at GitHub is called **GitHub-flavored Markdown**
```

tobyhodges commented on _episodes/02-markdown.md:
```diff
@@ -24,9 +24,220 @@ keypoints:
+> However, there are differences in the details and scope of the features offered.
+> To address this situation, the Markdown 
+> used at Github is called **Github-flavored Markdown**
+> and the one used at Gitlab is called **GitLab Flavored Markdown**.
```
```suggestion
> and the one used at GitLab is called **GitLab Flavored Markdown**.
```

tobyhodges commented on _episodes/02-markdown.md:
```diff
@@ -24,9 +24,220 @@ keypoints:
+> used at Github is called **Github-flavored Markdown**
+> and the one used at Gitlab is called **GitLab Flavored Markdown**.
+> - So be aware of
+>   - [Github-flavored Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)
```
```suggestion
>   - [GitHub-flavored Markdown](https://github.com/adam-p/markdown-here/wiki/Markdown-Cheatsheet)
```

tobyhodges reviewed (changes requested):
Fantastic! A few suggestions, and the `{: .markdown }` tags need changing, but otherwise we're good to go 👍

JulianKarlBauer committed:
Update _episodes/02-markdown.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

JulianKarlBauer committed:
Update _episodes/02-markdown.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

JulianKarlBauer committed:
Update _episodes/02-markdown.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

JulianKarlBauer committed:
Update _episodes/02-markdown.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

JulianKarlBauer committed:
Update _episodes/02-markdown.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

JulianKarlBauer committed:
Update _episodes/02-markdown.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

JulianKarlBauer committed:
Update _episodes/02-markdown.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

JulianKarlBauer committed:
Update _episodes/02-markdown.md

Co-authored-by: Toby Hodges <tbyhdgs@gmail.com>

JulianKarlBauer committed:
Resolve part of Toby's review: 

the lesson template (currently) has no highlighting rules for Markdown 
source. Best to change all these {: .markdown } tags to {: .source }

JulianKarlBauer committed:
Fix inconsistency

JulianKarlBauer committed:
Fix case of language R->r in solution block

tobyhodges commented:
Thank you, Julian!
